sailor/assetcentral/system.py

```diff
@@ -44,20 +44,111 @@ def get_property_mapping(cls):
+        """
+        name, order, id, model, row = [], [], [], [], []
+        r = 0
+        for child in component['childNodes']:
+            name.append(child['name'])
+            order.append(child['order'])
+            id.append(child['id'])
+            model.append(child['model'])
+            row.append(r)
+            r += 1
+        compdf = pd.DataFrame(list(zip(name, id, order, model, row)), columns=['name', 'id', 'order', 'model', 'row'])
+        compdf['one'] = 1
+        compdf.sort_values(by=['model', 'order'], inplace=True)
+        compdf['model_order'] = compdf.groupby(['model']).agg(rank=('one', 'cumsum'))
+        compdf.sort_values(by='order', inplace=True)
+        return compdf
+
+    @staticmethod
+    def _traverse_components(component, model_order, equipment_ids, system_ids):
+        """Traverse component structure recursively, starting from `component`."""
+        compd = {}
+        compd['key'] = (component['model'], model_order)
+        compd['id'] = component['id']
+        compd['name'] = component['name']
+        compd['order'] = component['order']
+        if component['order'] is not None:
+            if component['objectType'] == 'EQU':
+                equipment_ids.append(component['id'])
+            else:
+                system_ids.append(component['id'])
+        compd['object_type'] = component['objectType']
+        compdf = System._sort_children(component)
+        if len(compdf) > 0:
+            compd['child_list'] = []
+            for c in range(len(compdf)):
+                row = compdf.iloc[c]['row']
+                model_order = compdf.iloc[c]['model_order']
+                compd0, equipment_ids, system_ids = System._traverse_components(component['childNodes'][row],
+                                                                                model_order, equipment_ids, system_ids)
+                compd['child_list'].append(compd0)
+        return compd, equipment_ids, system_ids
+
+    def _update_components(self, component):
+        if component['object_type'] == 'EQU':
+            obj = self._equipments.filter(id=component['id'])[0]
+            component['indicators'] = self._indicators[component['id']]
+        else:
+            if component['id'] == self.id:
+                obj = self
+            else:
+                obj = self._systems.filter(id=component['id'])[0]
+        component['key'] = (obj.model_id, component['key'][1])
+        if 'child_list' in component.keys():
+            component['child_nodes'] = {}
+            component['child_order'] = []
+            for child in component['child_list']:
+                self._update_components(child)
+                component['child_nodes'][child['key']] = child
+                component['child_order'].append(component['child_nodes'][child['key']]['key'])
+                del component['child_nodes'][child['key']]['key']
+            del component['child_list']
```

Comment:
can this section be simplified somehow?
`child_list, child_nodes, child_order` do we really need all this?